src/DB/QueryWriterInterface.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\DB;
+
+interface QueryWriterInterface
+{
+    public function writeQuery(QueryInfo $info): string;
```

Comment:
__optional__

query is redundant as we can se that from the context so it could be just `write`.